### PR TITLE
[[ Docs ]] Make sure array variables are parsed and output as expected

### PR DESCRIPTION
**Pull request by livecodeali**

### CODE REVIEW

**Comment by peter-b** [comment]
Does this fix [bug 17521](http://quality.livecode.com/show_bug.cgi?id=17521)?

**Comment by livecodeali** [comment]
This + https://github.com/livecode/livecode-ide/pull/1272

File: ide-support/revdocsparser.livecodescript [hunk 1]
```diff
@@ -947,13 +947,16 @@ private function __elementRegex
...
-/*
-Utility function to extract any elements matching the docs element style
-*/
-function revDocsExtractElements pText
-   # Get the element-matching regex
+private function __arrayElementRegex
+   # Regex used to extract sub-array elements
+   # Essentially the same as the element regex except quotation marks are allowed
+   # as keys can obviously be string literals.
+   return "^ *([\w" & quote & "]+) *(?:\(( *(?:optional |in |out |inout )?<?\w*>?) *\))? *: *(.*)"
```
Comment (peter-b) [inline ide-support/revdocsparser.livecodescript:954]: Would it be useful to match multiple whitespace characters after "optional, "in" or "out"?  Something like `(?:(?:optional|in|out) *)`

Comment (livecodeali) [inline ide-support/revdocsparser.livecodescript:954]: Probably, but that should be a separate bug/enhancement request & pull request I think, given that it should also apply to the general element regex.

**Comment by peter-b** [comment]
@livecodeali Seems reasonable!

@livecode-vulcan review ok 178bd36

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok 178bd36754e6c1362b2b2918e9c111336180b86e

**Comment by livecode-vulcan** [comment]
😎 test success 178bd36754e6c1362b2b2918e9c111336180b86e
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/15)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/15)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/15)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/15)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/15)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/15)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/15)

**Pull request merged**